engine/src/witness/sol.rs

```diff
@@ -175,6 +183,41 @@ impl VoterApi<SolanaLiveness> for SolanaLivenessVoter {
 	}
 }
 
+#[allow(dead_code)]
+#[derive(Clone)]
+struct SolanaVaultSwapsVoter {
+	client: SolRetryRpcClient,
+}
+
+#[async_trait::async_trait]
+impl VoterApi<SolanaVaultSwapTracking> for SolanaVaultSwapsVoter {
+	async fn vote(
+		&self,
+		settings: <SolanaVaultSwapTracking as ElectoralSystem>::ElectoralSettings,
+		properties: <SolanaVaultSwapTracking as ElectoralSystem>::ElectionProperties,
+	) -> Result<
+		<<SolanaVaultSwapTracking as ElectoralSystem>::Vote as VoteStorage>::Vote,
+		anyhow::Error,
+	> {
+		program_swaps_witnessing::get_program_swaps(
+			&self.client,
+			settings.swap_endpoint_data_account_address,
+			properties
+				.witnessed_open_accounts
+				.into_iter()
+				.map(|VaultSwapAccountAndSender { vault_swap_account, .. }| vault_swap_account)
+				.collect(),
+			properties.closure_initiated_accounts,
```

Comment:
doesn't look like we use the `swap_sender` from here either? By removing the sender here, it could allow us to simplify the code in `get_changed_program_swap_accounts` even further than my suggestion by using set operators instead of filter collect()

Reply:
good point, will do

Reply:
This conversation was resolved but nothing was changed here?

Reply:
I did simplify that function but that was for another comment and I missed this one.
I think not passing the sender downstream might be not be very clean to implement as we do require the sender for each of those accounts in the vote. Maybe there is a nice way to do it in Rust that I'm missing.
What I think could be a better option is to not require the sender address when voting on a closed account. We are anyway not doing anything with it since we report just the sender address that the SC has told the engine to witness.
I see how that might require some update in the pallet as we currently store those accounts in a `BTreeSet<(Address, Sender)>` in `closure_initiated_accounts`. We could technically also not store the sender in there and just have a `BTreeSet<Address>` as we don't need the anymore once the closure transaction has been initiated.
What do you think? @kylezs @ramizhasan111

Reply:
Fair, probably simpler to just leave it as is I think